src/components/aside-bar/AsideBar.jsx

```diff
@@ -1,68 +1,150 @@
 import "./aside-bar.css";
 import { Button } from "../index";
+import { useProductContext } from "../../context";
+import { useState } from "react";
 
 const AsideBar = () => {
+  const [key, setKey] = useState(0);
+  // Product Context
+  const {
+    state: { sortBy },
+    dispatch,
+  } = useProductContext();
+
+  // Reset Function
+  const resetFunction = () => {
+    setKey((prev) => prev + 1);
+    dispatch({ type: "SHIRTS", payload: false });
+    dispatch({ type: "YOGA_MATS", payload: false });
+    dispatch({ type: "PANTS", payload: false });
+    dispatch({ type: "SORT_BY", payload: true });
+    dispatch({ type: "RATING", payload: 0 });
```

Comment:
Instead of doing this, you can directly specify these reset actions in the "clear" case in reducer, and on the button click just dispatch to "clear".

Reply:
thankyou